Source/WTF/wtf/text/StringView.cpp

```diff
@@ -447,6 +447,35 @@ int codePointCompare(StringView lhs, StringView rhs)
     return codePointCompare(lhs.characters16(), lhs.length(), rhs.characters16(), rhs.length());
 }
 
+template<typename CharacterType> static String makeStringBySimplifyingNewLinesSlowCase(const String& string, unsigned firstCarriageReturn)
+{
+    unsigned length = string.length();
+    unsigned resultLength = firstCarriageReturn;
+    auto* characters = string.characters<CharacterType>();
+    CharacterType* resultCharacters;
+    auto result = String::createUninitialized(length, resultCharacters);
+    memcpy(resultCharacters, characters, firstCarriageReturn * sizeof(CharacterType));
+    for (unsigned i = firstCarriageReturn; i < length; ++i) {
+        if (characters[i] != '\r')
```

Comment:
I have a slight concern about whether the existing function implemented the correct algorithm. Do we know for certain what normalization is supposed to be done? I've seen some weird newline combinations in the past (`\n\r`, `\r\r\n` are treated as a single newline by some software).

Reply:
I am certain the function is doing the correct normalization. I researched this originally and still remember learning this was exactly what is needed.